src/Middleware/MiddlewareAnalysis/test/MiddlewareAnalysisTests.cs

```diff
@@ -53,6 +53,6 @@ public async Task ExceptionWrittenToDiagnostics()
         Assert.Equal("Microsoft.AspNetCore.MiddlewareAnalysis.MiddlewareAnalysisTests+<>c", listener.MiddlewareException[0]);
         // reversed "Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware"
         Assert.Equal(1, listener.MiddlewareFinished.Count);
-        Assert.Equal("Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware", listener.MiddlewareFinished[0]);
+        Assert.Equal("Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl", listener.MiddlewareFinished[0]);
```

Comment:
We should try to preserve the category name of `Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware`

Reply:
We are not changing the category name:

https://github.com/dotnet/aspnetcore/blob/988a20e1a071d9aa3c1f75814724200bb22b504f/src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddlewareImpl.cs#L75

However, this `AnalysisMiddleware` uses the type name to create those events :(. I tried but I am not sure we can avoid this.